[Kee Seow Han] iP

KSHan29 opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/264).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method: 
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

KSHan29 committed:
Duke Level-1

KSHan29 committed:
Level-2

KSHan29 committed:
Level-3

KSHan29 committed:
Level-4

KSHan29 committed:
Level-5

KSHan29 committed:
Level-6

KSHan29 committed:
A-TextUiTesting

KSHan29 committed:
Fix coding style

KSHan29 committed:
Fix deadline bug

KSHan29 committed:
Update testing inputs and outputs

KSHan29 committed:
Implement read from disk on start and on add task

KSHan29 committed:
Fix numbered tasks bug

KSHan29 committed:
Implement disk read/write for methods except mark/unmark

KSHan29 committed:
Implement disk write/read for mark and unmark

KSHan29 committed:
Implement LocalDate for Events and Deadlines

KSHan29 committed:
no message

KSHan29 committed:
Merge branch 'branch-Level-8'

KSHan29 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Event.java

KSHan29 committed:
Fix bugs from merge

KSHan29 committed:
Implement more OOP

KSHan29 committed:
Place files in package

KSHan29 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

KSHan29 committed:
A-Gradle

KSHan29 committed:
Add unit tests

KSHan29 committed:
Package app as JAR file

KSHan29 committed:
Add JavaDoc for classes and methods

KSHan29 committed:
Revert "Add JavaDoc for classes and methods"

This reverts commit 71e98551b647ca25b26250e7c16176c644fab130.

KSHan29 committed:
Fix coding style violations

KSHan29 committed:
Add method Find to search Task list

KSHan29 committed:
Add newline to end of files

KSHan29 committed:
Merge branch 'branch-Level-9'

KSHan29 committed:
Merge branch 'branch-A-CodingStandard'

KSHan29 committed:
Add JavaFx code for build.gradle

KSHan29 committed:
Merge with master branch

KSHan29 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Merge with master branch

# Conflicts:
#	src/main/java/duke/Storage.java

KSHan29 committed:
Revert "Merge with master branch"

This reverts commit a2544e7001c2a2d1396263627f4f38de64c98046.

KSHan29 committed:
Merge with branch-A-JavaDoc

KSHan29 committed:
Add StyleCheck using gradle

KSHan29 committed:
Revert "Add StyleCheck using gradle"

This reverts commit 8120621139f3849c2b8fba63118620a6427d8f13.

KSHan29 committed:
Add CheckStyle

KSHan29 committed:
Merge branch 'branch-A-CheckStyle'

KSHan29 committed:
Improve OOP

KSHan29 committed:
Refactor files

KSHan29 committed:
Implement JavaFX for Duke GUI

KSHan29 committed:
Merge branch 'branch-Level-10'

KSHan29 committed:
Update configurations for jar file

chm780 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Handles exceptions in duke.Duke program relating to invalid arguments.
+ */
+public class DukeException extends IllegalArgumentException {
```
Good that you are specific which exception you extended from instead of just Exception :)

chm780 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+import duke.commands.Command;
+import javafx.scene.image.Image;
+
+/**
```
Good coding habit to document class 👍

chm780 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,115 @@
+ * Handles interaction with user.
+ */
+public class Ui {
+//    /**
```
Perhaps can remove these unused lines? :)

chm780 commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,68 @@
+            tasks.addTask(task);
+            storage.addTaskToDisk(task.taskMemo() + System.lineSeparator());
+            return ui.printAddTask(task, tasks.getTaskListSize());
+        } catch (IOException e) {
```
Good consideration of possible exceptions thrown

chm780 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+     * @param filePath filePath of file to store Task information.
+     * @param tempFilePath filePath of temporary file to store information for rewriting.
+     */
+    public Duke(String filePath, String tempFilePath) {
```
Well usage of OOP 👍

autumn-sonata commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import duke.*;
```
Try not to import using the `*`.

autumn-sonata commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,204 @@
+package duke;
+
+import duke.tasks.Deadline;
```
Good job on cleanly separating the imports, it is easy to read. 👍

autumn-sonata commented on src/test/java/duke/EventTest.java:
```diff
@@ -0,0 +1,23 @@
+    }
+
+    @Test
+    void taskMemo() {
```
Maybe the function name could be something like this to be consistent :)
```suggestion
public void testTaskMemo() {
```

autumn-sonata commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,115 @@
+     * Greets user.
+     */
+    public static String showWelcome() {
+//        String logo = " ____        _        \n"
```
Might want to remove these lines from the code base

autumn-sonata commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,204 @@
+     * @param isDone the new status of the task.
+     * @throws IOException if unable to write to file.
+     */
+    public void setTaskStatusOnDisk(int taskNumber, boolean isDone)throws IOException {
```
Slight nit, add a space between the parameters and throws.
```suggestion
public void setTaskStatusOnDisk(int taskNumber, boolean isDone) throws IOException {
```

autumn-sonata reviewed (commented):
Good job on following the Java coding standards! 💯

glemenneo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,204 @@
+            hardDiskTasks.createNewFile();
+            tempTasks.createNewFile();
+        } catch (IOException | SecurityException e) {
+            System.out.println("     " + e.getMessage());
```
Consider making use of \t escape sequence to insert a tab.

glemenneo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,204 @@
+    }
+
+    /**
+     * Loads Tasks previously added from memory and adds into an ArrayList.
```
Lots of JavaDocs and detailed comments for easier understanding which is very helpful!

glemenneo commented on src/main/java/duke/commands/ExitCommand.java:
```diff
@@ -0,0 +1,38 @@
+    /**
+     * ExitCommand constructor.
+     */
+    public ExitCommand() {
```
Why declare a constructor that will be the same as the default one? Just a small nitpick.

KSHan29 committed:
Fix coding style violations

KSHan29 committed:
Add JAR file for JavaFX

KSHan29 committed:
Add assertions to check for assumption errors

KSHan29 committed:
Improve code quality

Some code is lengthy and have undescriptive method names.

The code is hard to understand which may result in poor maintainability in the future.

Let's improve the readability of the code by refactoring them using:
*renaming
*method extraction

*Renaming ensures variables are descriptive so ther programmers know what is the purpose of the variable.
*Method extraction ensures a method is not overly complicated with different levels of abstraction performed in the same method.

KSHan29 committed:
Merge pull request #1 from KSHan29/branch-A-CodeQuality

Improve code quality

KSHan29 committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/duke/Storage.java

KSHan29 committed:
Add .yml file

KSHan29 committed:
Create gradle.yml

KSHan29 committed:
no message

KSHan29 committed:
Revert "no message"

This reverts commit 8c96739cbad6e90d5b35b855b782e567542bd4f2.

KSHan29 committed:
Revert "Add .yml file"

This reverts commit 92a2506f860104486277c8c9bb5f07212445d541.

KSHan29 committed:
Fix coding violations

KSHan29 committed:
Merge branch 'branch-C-DetectDuplicate'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/commands/TaskCommand.java

KSHan29 committed:
Merge branch 'branch-C-DetectDuplicate'

KSHan29 committed:
Fix coding style violations

KSHan29 committed:
Fix coding style violations

KSHan29 committed:
Update gradle.yml

Remove I/O redirection tests

KSHan29 committed:
Storage class: remove temporary file usage

Storage class uses temporary file for store data to rewrite Task memo file.

Temporary file is unnecessary and the file writing may be slow.

Let's use a String to keep track of what to overwrite and remove usage of the temporary file.

Improves performance with less I/O operation and less likely to encounter exceptions from I/O operations.

KSHan29 committed:
Merge branch 'master' of https://github.com/KSHan29/ip

KSHan29 committed:
Fix error in Task

KSHan29 committed:
Fix coding style violations

KSHan29 committed:
docs/README.md: Update User Guide

KSHan29 committed:
Improve GUI

KSHan29 committed:
Add image to commit

KSHan29 committed:
Add Ui.png

KSHan29 committed:
Add support for JAR file for Windows

KSHan29 committed:
Update User Guide

KSHan29 committed:
Update User Guide

KSHan29 committed:
Update User Guide

KSHan29 committed:
Fix User Guide hyperlink

KSHan29 committed:
Fix User Guide summary table format

KSHan29 committed:
Update User Guide summary table format

KSHan29 committed:
Update User Guide summary table format

KSHan29 committed:
Fix User Guide summary table format

KSHan29 committed:
Update README.md

KSHan29 committed:
Update User Guide

KSHan29 committed:
Merge from git

KSHan29 committed:
Update README.md

KSHan29 committed:
Update README.md

KSHan29 committed:
Update README.md

KSHan29 committed:
Update README.md

KSHan29 committed:
Update README.md

Commit:
Add Project image
